tests/test_wsireader.py

```diff
@@ -2979,3 +2987,119 @@ def test_fsspec_reader_open_pass_empty_json(tmp_path: Path) -> None:
     json_path.write_text("{}")
 
     assert not FsspecJsonWSIReader.is_valid_zarr_fsspec(str(json_path))
+
+
+def test_read_rect_transformedreader_svs_baseline(
+    sample_svs: Path, remote_sample: Callable, tmp_path: Path
+) -> None:
+    """Test TransformedWSIReader.read_rect with an SVS file at baseline."""
+    wsi = wsireader.TransformedWSIReader(
+        sample_svs, target_img=sample_svs, transform=np.eye(3)
+    )
+    location = SVS_TEST_TISSUE_LOCATION
+    size = SVS_TEST_TISSUE_SIZE
+    im_region = wsi.read_rect(location, size, resolution=0, units="level")
+
+    assert isinstance(im_region, np.ndarray)
+    assert im_region.dtype == "uint8"
+    assert im_region.shape == (*size[::-1], 3)
+
+    fixed_info = wsi.info
+    wsi2 = wsireader.TransformedWSIReader(
+        sample_svs, target_img=sample_svs, transform=None, fixed_info=fixed_info
+    )
+    im_region_2 = wsi2.read_rect(location, size, resolution=0, units="level")
+
+    assert np.array_equal(im_region, im_region_2)
+
+    # Now test MHA displacement field
+    wsi3 = wsireader.TransformedWSIReader(
+        sample_svs,
+        target_img=sample_svs,
+        transform=remote_sample("reg_disp_mha_example"),
+    )
+    im_region_3 = wsi3.read_rect(location, size, resolution=0, units="level")
+
+    # We don't expect arrays to be the same, but dimensions should be
+    assert im_region.shape == im_region_3.shape
+
+    # Now test NPY affine transformation
+    wsi4 = wsireader.TransformedWSIReader(
+        sample_svs,
+        target_img=sample_svs,
+        transform=remote_sample("reg_affine_npy_example"),
+    )
+    im_region_4 = wsi4.read_rect(location, size, resolution=0, units="level")
+
+    # We don't expect arrays to be the same, but dimensions should be
+    assert im_region.shape == im_region_4.shape
```

Comment:
Can we also test the values or just shape is enough?

Reply:
Here, we haven't performed the test with the correct MHA files as didn't want to add lots of new slides to the server. So think shape alone is reasonable.